src/fides/api/ctl/migrations/versions/643249f65453_case_insensitive_usernames_delete_.py

```diff
@@ -0,0 +1,48 @@
+"""case_insensitive_usernames_delete_duplicates
+
+Revision ID: 643249f65453
+Revises: 392992c7733a
+Create Date: 2023-02-01 22:22:38.055862
+
+"""
+from alembic import op
+from sqlalchemy.orm.session import Session
+from sqlalchemy.sql import text
+
+from fides.lib.models.fides_user import FidesUser
+from fides.lib.models.fides_user_permissions import FidesUserPermissions
+
+# revision identifiers, used by Alembic.
+revision = "643249f65453"
+down_revision = "392992c7733a"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+
+    # Found here in the 24.2.2.4. Nondeterministic Collations section:
+    # https://www.postgresql.org/docs/current/collation.html
+    op.execute(
+        "CREATE COLLATION IF NOT EXISTS case_insensitive (provider = icu, locale = 'und-u-ks-level2', deterministic = false);"
+    )
+    session = Session(bind=op.get_bind())
+    conn = session.connection()
+    statement = text(
+        """SELECT DISTINCT ON (username) username collate case_insensitive, created_at, id
```

Comment:
I ran this query using raw sql because I was having a pretty hard time getting the ORM to generate it correctly. It was always pretty close but never worked quite right.

Reply:
^^^ Have experienced this before! Especially when Fideslang changes

Reply:
That makes a lot sense. I'll keep using raw sql in the future